caffe2/utils/threadpool/pthreadpool-cpp.cc

```diff
@@ -2,8 +2,42 @@
 #include <caffe2/utils/threadpool/thread_pool_guard.h>
 #include <c10/util/Exception.h>
 
+#include <atomic>
+
+namespace {
+// Number of threads in the thread-pool before fork
+std::atomic<int> num_threads_before_fork{-1};
+
+} // namespace
+
 namespace caffe2 {
 
+// Handler used by pthread_atfork that's executed before fork starts processing.
+// Saves the value of the number of threads prior to fork, so that the Caffe2
+// threadpool can be destroyed, and restored via after_fork handler, after fork.
+// It's done in order to prevent segfaults in worker processes, which otherwise
+// segfault, as they try to handle inherited data-structures of parent's Caffe2
+// thread-pool, although those threads don't exist in child processes.
+// Ref: https://github.com/pytorch/pytorch/issues/54752#issuecomment-810315302
+void before_fork() {
```

Comment:
These handlers should probably be `static`. I'll modify after CI checks would complete.